src/mono/mono/mini/aot-runtime.c

```diff
@@ -2443,24 +2443,20 @@ load_container_amodule (MonoAssemblyLoadContext *alc)
 	if (!container_assm_name || container_amodule)
 		return;
 
-	char *local_ref = container_assm_name;
+	// Create a fake MonoAssembly/MonoImage to retrieve its AOT module.
+	// Container MonoAssembly/MonoImage shouldn't be used during the runtime.
+	MonoAssembly *assm = g_new0 (MonoAssembly, 1);
+	assm->image = g_new0 (MonoImage, 1);
+	assm->image->dynamic = 0;
+	assm->image->alc = alc;
+	assm->aname.name = container_assm_name;
```

Comment:
Couple of suggestions:
1. Allocate the MonoImage from some mempool... maybe from the ALC
2. I think you need to increment the refcount of the MonoImage.

In general, this should be like a fake version of 
```c
MonoAssembly *
mono_assembly_request_load_from (MonoImage *image, const char *fname,
				 const MonoAssemblyLoadRequest *req,
				   MonoImageOpenStatus *status)
```

Reply:
Can `mono_assembly_addref` be used to increment reference count?

Do you think that something else should be added from `mono_assembly_request_load_from` and `do_mono_image_open`?